### PR TITLE
Emailer

### PR DESCRIPTION
**Pull request by dxa4481**

### CODE REVIEW

File: hflossk/site.py [hunk 1]
```diff
@@ -34,6 +43,50 @@ def inject_yaml():
...
+                if(second_day):
+	            due_dates.append(current_week)
+                else:
+                    due_dates.append(current_week + two_days)
+            second_day = True
+        current_week = current_week + one_week
+    for due_date in due_dates:
+        send_time = due_date - datetime.now().date() - two_days
+	if send_time.total_seconds() > 0:
+	    t = threading.Timer(send_time.total_seconds(), send_email)
```
Comment (dxa4481) [inline hflossk/site.py]: Hi @Qalthos, @decause said you would be the right person to ping. I want fire off an email on a fixed schedule, is this threading.Timer an appropriate way to do this?

**Comment by decause** [comment]
@dxa4481 says that the functions run, and nothing catastrophic happens.

I'm just concerned that some unsuspecting HFOSS student will want email updates, add their username and password for their email account, unwittingly `git add -a` and then compromise their email account :-1: Perhaps we could use Environment Variables instead of plain text inside of `site.py`?

Either way, a code review is needed. @ryansb, or @Qalthos , possibly @ralphbean, could you just give this a quick sanity check?

Thanks All!

File: hflossk/site.py [hunk 2]
```diff
@@ -34,6 +43,50 @@ def inject_yaml():
...
+        schedule = yaml.load(schedule_yaml)
+    return schedule
+
+def homework_reminder():
+    site_config = inject_yaml()
+    schedule = get_schedule()
+    start_date = site_config['course']['start']
+    due_dates = []
+    one_week = datetime(1,1,8) - datetime(1,1,1)
+    two_days = datetime(1,1,3) - datetime(1,1,1)
```
Comment (Qalthos) [inline hflossk/site.py]:
```
one_week = timedelta(weeks=1)
two_days = timedelta(days=2)
```

I tend to just use them inline for clarity unless it's some weird period with more than two parameters

**Comment by Qalthos** [comment]
It doesn't look like it will kill anyone or try to take over the world. It might even work.

This seems wildly out of scope for the site, though. At the very least, add a config flag to turn the emails off?

**Comment by ryansb** [comment]
Agree w/ @Qalthos. Seems insanely out of scope, and I feel like a better way would be to generate an iCal file for download w/ all homework assignments.

**Comment by ryansb** [comment]
I might add that since this runs on openshift, _someone_ would have to commit their SMTP creds, which seems non-great.

**Comment by dxa4481** [comment]
I mean I can make a gmail account for the express purpose of this application.

**Comment by ralphbean** [comment]
I'd like to point out that in principle, this is the coolest thing that has ever happened.  A student wrote code for the course infrastructure to remind them about homeworks.

That said, I think @ryansb's technical concern about committing SMTP creds is right on.  @dxa4481, even if you make a gmail account just for the purpose of this application, I'll then go take those creds and sell them to my friend.  He's a Nigerian prince and has this crazy financial situation.

Exporting a .iCal file seems nice.  It would serve the same basic purpose (although it assumes that students are already using calendaring which may not be the case).

**Comment by ryansb** [comment]
Suggestion: if you change it to read creds from a file (one not stored in the repo) and have @decause copy it manually to OpenShift, I'd :+1: that.

**Comment by ryansb** [comment]
And if it doesn't find the file, just gracefully turn off email functionality.

**Comment by dxa4481** [comment]
What about a config file in the repo that is just a part of the .gitignore

**Comment by dxa4481** [comment]
@ryansb

**Comment by ryansb** [comment]
Where on the file system the configuration file lives isn't too important, as long as it never gets committed to the repo.

**Comment by ralphbean** [comment]
> What about a config file in the repo that is just a part of the .gitignore

That would do it.

**Comment by dxa4481** [comment]
@ralphbean okay so I wanted to commit an example config file, and then add that file to the .gitignore but it looks like git won't let you ignore tracked files. 

So what I did was I updated the readme to say "you need to create this file if you want email notifications", and then added the filename to the gitignore. Then in site.py I wrapped the import in a try catch and if the config file doesn't exist it sets a boolean to not bother with any of the email stuff.

**Comment by ryansb** [comment]
Well done, @dxa4481.

Looks good to me. :+1:

**Comment by Nolski** [comment]
@decause ping. Any reason this was closed? This seems like it could solve many of the problems in issue #495

**Pull request closed without merging**